### PR TITLE
Create samtools_stats

### PR DESCRIPTION
**Pull request by cjuigne**

### CODE REVIEW

File: samtools/samtools_stats [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+- samtools
+- stats
+requirements:
+- class: DockerRequirement
+  dockerPull: biocontainers/samtools:v1.7.0_cv3
+
+inputs:
+  input_file:
+    type: File
+  #   format:
```
Comment (mr-c) [inline samtools/samtools_stats]:
```suggestion
     format: edam:format_2572  # BAM
```

**Review by mr-c — CHANGES_REQUESTED** [review]
One small request about the `format` otherwise this is great, thanks!

File: samtools/samtools_stats [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      prefix: --coverage
+    doc: "Set coverage distribution to the specified range (MIN, MAX, STEP all given as integers) [1,1000,1]"
+  remove_dups:
+    type: boolean?
+    doc: "Exclude from statistics reads marked as duplicates"
+    inputBinding:
+      prefix: --remove_dups
+
+  required_flag:
+    type: string?
```
Comment (mr-c) [inline samtools/samtools_stats]:
```suggestion
    type:
      - string
      - int
      - "null"|
```

Comment (mr-c) [reply samtools/samtools_stats]: The same for the other `STR|INT` items, thanks!

File: samtools/samtools_stats [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  cov_threshold:
+    type: int?
+    doc: "Only bases with coverage above this value will be included in the target percentage computation [0] "
+    inputBinding:
+      prefix: -g
+  threads:
+    type: int?
+    doc: "Number of input/output compression threads to use in addition to main thread [0]."
+    inputBinding:
+      prefix: --threads
```
Comment (mr-c) [inline samtools/samtools_stats]:
```suggestion

arguments:
  - prefix: --threads
    valueFrom: $(runtime.cores)  
```

File: samtools/samtools_stats [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+# -X
+#     If this option is set, it will allows user to specify customized index file location(s) if the data folder does not contain any index file. Example usage: samtools stats [options] -X /data_folder/data.bam /index_folder/data.bai chrM:1-10
+
+outputs:
+  stats:
+    type: File
```
Comment (mr-c) [inline samtools/samtools_stats]:
```suggestion
    type: File
    format: iana:text/plain
```

File: samtools/samtools_stats [hunk 5]
```diff
@@ -0,0 +1,142 @@
+class: CommandLineTool
+cwlVersion: v1.1
```
Comment (mr-c) [inline samtools/samtools_stats]:
```suggestion
cwlVersion: v1.0
```
Since I see no v1.1 specific features here

**Comment by cjuigne** [comment]
@mr-c Sorry I forgot to include the format specifications, thank you for your help!

**Pull request closed without merging**